src/default-directory-searcher.coffee

```diff
@@ -0,0 +1,107 @@
+{EventEmitter} = require 'events'
+Task = require './task'
+
+# Maintain a queue of ids of searches to run. When a search is complete, the active
+# search is cleared and the next search (if any) in the queue is notified to run.
+# This ensures there is at most one scan-handler task running at a time.
+searchQueue = []
+nextId = 1
+activeSearchId = 0
+emitter = new EventEmitter
```

Comment:
OK, let me try multiple directory support. This should be easier to implement now that the return value of a `DirectorySearcher` is only required to have `then()` and `cancel()` as methods. (This was more difficult when it was also responsible for `onDidMatch`, `onDidPathsSearched`, etc.)